lib/sycamore/sycamore/evaluation/evaluate.py

```diff
@@ -0,0 +1,127 @@
+from typing import Dict, List, Union
+from abc import ABC, abstractmethod
+
+import os
+from sycamore.reader import DocSetReader
+from sycamore.data import Element
+from sycamore.evaluation.pipeline import EvaluationPipeline
+from sycamore.evaluation import EvaluationDataPoint
+from sycamore.evaluation.metrics import document_retrieval_metrics, rouge_metrics
+from sycamore import Context
+
+
+class Assessment(ABC):
+
+    @abstractmethod
+    def run_evaluation(self, ctx: Context, index: str, **kwargs):
+        pass
+
+    def __call__(self, context: Context, index: str, **kwargs):
+        return self.run_evaluation(context, index, **kwargs)
+
+
+class QualityAssessment(Assessment):
+    def __init__(self, GT_path: str, rag_config: Dict, **kwargs):
+        self.user = os.environ.get("USER", os.environ.get("USERNAME"))
+        self.GT_path = GT_path
+        self.rag_config = rag_config
+        self.os_client_args = kwargs.get("os_client_args", "")
+        self.metrics = kwargs.get("metrics", [document_retrieval_metrics, rouge_metrics])
+        self.custom_question_augmentation = kwargs.get("custom_question_augmentation", {})
+        self.question_augmentation_filter = kwargs.get("question_augmentation_filter", "")
+
+    @staticmethod
+    def create_evaluation_datapoint(
+        json_dict: Dict, custom_question_augmentation: str = "{}", question_augmentation_filter: str = ""
+    ):
+        result = []
+        assert json_dict is not None
+        assert isinstance(json_dict, dict)
+        for datapoint in json_dict["data"][:]:
+            document = EvaluationDataPoint()
+            document.raw = datapoint
+            document.ground_truth_answer = datapoint["Answer"]
+            document.filters = datapoint.get("Filters", None)
+            if document.filters:
+                for filter in document.filters.keys():
+                    document.filters[filter] = datapoint.get("Filters").get(filter)
+                document.question = custom_question_augmentation.format(
+                    document.question, document.filters.get(question_augmentation_filter)
+                )
+            else:
+                document.filters = {}
+                document.question = custom_question_augmentation.format(document.question, "")
```

Comment:
suggestion: turn this into a map function. e.g.
```python
def add_filters_to_question(datapoint: Document) -> Document:
    assert isinstance(datapoint, EvaluationDataPoint)
    # add a bunch of stuff to the datapoint
    return datapoint

# ...
# then in run_evaluation
    input_docset = ctx.read.json(...).map(add_filters_to_question)
```
Can do a similar thing with the SearchContexts

Reply:
added a function to call using map on input json